diff --git a/CMake/PackageConfig.cmake b/CMake/PackageConfig.cmake
index ab4e3f4..986efdf 100644
--- a/CMake/PackageConfig.cmake
+++ b/CMake/PackageConfig.cmake
@@ -4,13 +4,13 @@ add_library(HighFive INTERFACE)
 target_include_directories(HighFive INTERFACE
   "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include>")
-target_include_directories(HighFive SYSTEM INTERFACE ${HDF5_INCLUDE_DIRS})
+target_include_directories(HighFive SYSTEM INTERFACE "$<BUILD_INTERFACE:${HDF5_INCLUDE_DIRS}>")
 target_link_libraries(HighFive INTERFACE ${HDF5_LIBRARIES})
 target_compile_definitions(HighFive INTERFACE ${HDF5_DEFINITIONS})
 
 # MPI
 if(HIGHFIVE_PARALLEL_HDF5 OR HDF5_IS_PARALLEL)
-  target_include_directories(HighFive SYSTEM INTERFACE ${MPI_CXX_INCLUDE_PATH})
+  target_include_directories(HighFive SYSTEM INTERFACE "$<BUILD_INTERFACE:${MPI_CXX_INCLUDE_PATH}>")
   target_link_libraries(HighFive INTERFACE ${MPI_CXX_LIBRARIES})
   if(CMAKE_VERSION VERSION_LESS 3.13)
     target_link_libraries(HighFive INTERFACE ${MPI_CXX_LINK_FLAGS})
@@ -21,7 +21,7 @@ endif()
 
 # BOOST
 if(USE_BOOST)
-  target_include_directories(HighFive SYSTEM INTERFACE ${Boost_INCLUDE_DIR})
+  target_include_directories(HighFive SYSTEM INTERFACE "$<BUILD_INTERFACE:${Boost_INCLUDE_DIR}>")
   target_compile_definitions(HighFive INTERFACE -DH5_USE_BOOST)
   target_link_libraries(HighFive INTERFACE ${Boost_SERIALIZATION_LIBRARIES})
 endif()
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 29a58c1..f61c2f2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,5 +102,3 @@ if(HIGHFIVE_UNIT_TESTS)
   enable_testing()
   add_subdirectory(tests/unit)
 endif()
-
-add_subdirectory(doc)
